### PR TITLE
Add "deactivate" to bulk user upload

### PR DESCRIPTION
**Pull request by proteusvacuum**
http://manage.dimagi.com/default.asp?175543#975261
http://dimagi.uservoice.com/forums/194748-user-management/suggestions/3706110-bulk-upload-of-users-should-allow-archiving-of-use

Adds an `is_active` column to bulk user upload so that users can be bulk-deactivated (or activated)

code buddy @biyeun 
( @dannyroberts FYI https://github.com/dimagi/commcare-hq/commit/67bb7a431e041f34c5615a849e7565095b4e8c62 )

### CODE REVIEW

File: corehq/apps/users/dbaccessors/all_commcare_users.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+def get_all_commcare_users_by_domain(domain):
+    """Returns all CommCareUsers by domain regardless of their active status"""
+    from corehq.apps.users.models import CommCareUser
+    key = [domain, CommCareUser.__name__]
+
+    return [CommCareUser.wrap(user['doc']) for user
```
Comment (czue) [inline corehq/apps/users/dbaccessors/all_commcare_users.py]: this should probably use [iter_docs](https://github.com/dimagi/dimagi-utils/blob/master/dimagi/utils/couch/database.py#L53) - we've observed that performance is significantly better if you do this for anything that can return more than a hundred or so results

File: corehq/apps/users/bulkupload.py [hunk 2]
```diff
@@ -629,7 +648,7 @@ def _load_memoizer(domain):
 
     user_headers, user_rows = parse_users(
         group_memoizer,
-        CommCareUser.by_domain(domain),
+        get_all_commcare_users_by_domain(domain),
```
Comment (czue) [inline corehq/apps/users/bulkupload.py]: (non-actionable) it could be worth also porting `CommCareUser.by_domain` to call a `dbaccessors` function (I see it's a bit more complicated than what you wrote so would be ok being told this is out of scope)

File: corehq/apps/users/bulkupload.py [hunk 3]
```diff
@@ -363,6 +364,20 @@ def _set_progress(progress):
                 'username': raw_username(username) if username else None,
                 'row': row,
             }
+
+            is_active = row.get('is_active')
+            if type(is_active) in (unicode, str):
+                active_string = is_active.lower().strip()
+                if active_string in ('true', 'false'):
+                    is_active = active_string == 'true'
```
Comment (czue) [inline corehq/apps/users/bulkupload.py]: (not blocking) could use `string_to_boolean` which is slightly more permissive

File: corehq/apps/users/dbaccessors/all_commcare_users.py [hunk 4]
```diff
@@ -0,0 +1,12 @@
+def get_all_commcare_users_by_domain(domain):
```
Comment (czue) [inline corehq/apps/users/dbaccessors/all_commcare_users.py]: might be worth adding a quick test for this

File: corehq/apps/users/dbaccessors/all_commcare_users.py (hunk 1 above)
Comment (czue) [inline corehq/apps/users/dbaccessors/all_commcare_users.py]: (though I see that the previous implementation also doesn't do this)

**Comment by czue** [comment]
my comments aren't blocking - happy for @biyeun or @dannyroberts to merge if they're happy

File: corehq/apps/users/bulkupload.py [hunk 5]
```diff
@@ -363,6 +364,20 @@ def _set_progress(progress):
                 'username': raw_username(username) if username else None,
                 'row': row,
             }
+
+            is_active = row.get('is_active')
+            if type(is_active) in (unicode, str):
```
Comment (millerdev) [inline corehq/apps/users/bulkupload.py]: (not blocking) A better way to do this might be `isinstance(is_active, basestring)` since that will catch subclasses of `str` and `unicode` as well.

**Comment by dannyroberts** [comment]
@proteusvacuum Looks good to me, just gonna give you a chance to read over comments in case anything catches your eye. Give me the thumbs up when you're done and I'll merge

**Comment by proteusvacuum** [comment]
@dannyroberts :+1: when tests pass

File: corehq/apps/users/bulkupload.py (hunk 3 above)
Comment (proteusvacuum) [inline corehq/apps/users/bulkupload.py]: sweet. didn't know that existed.

File: corehq/apps/users/bulkupload.py (hunk 2 above)
Comment (proteusvacuum) [inline corehq/apps/users/bulkupload.py]: Yeah, think that might be out of scope for this ticket.

File: corehq/apps/users/dbaccessors/all_commcare_users.py (hunk 1 above)
Comment (proteusvacuum) [inline corehq/apps/users/dbaccessors/all_commcare_users.py]: :+1:

File: corehq/apps/users/bulkupload.py (hunk 5 above)
Comment (proteusvacuum) [inline corehq/apps/users/bulkupload.py]: :+1:
